src/mobster/cmd/generate/oci_image/base_images_dockerfile.py

```diff
@@ -0,0 +1,391 @@
+        components.append(component)
+    return components
+
+
+async def _get_spdx_packages_from_base_images(
+    base_images_refs: list[str | None], base_images: dict[str, Image]
+) -> tuple[list[Package], list[Annotation]]:
+    """
+    Transforms the list of base images and their mapping to
+    an Image object into a list of SPDX Packages.
+    Args:
+        base_images_refs (list[str | None]):
+            list of image references, the last one is the parent image.
+        base_images (dict[str, Image]):
+            mapping of those references to Image objects.
+
+    Returns:
+        list[spdx_tools.spdx.model.Package]:
+            List of SPDX packages to be added to an SBOM.
+    """
+    packages = []
+    result_annotations = []
+    for image, annotations in await _get_images_and_their_annotations(
+        base_images_refs, base_images
+    ):
+        package = get_image_package(image, spdx_id=image.propose_spdx_id())
+        for annotation in annotations:
+            result_annotations.append(
+                Annotation(
+                    spdx_id=package.spdx_id,
+                    annotation_type=AnnotationType.OTHER,
+                    annotation_date=datetime.now(timezone.utc),
+                    annotator=Actor(
+                        actor_type=ActorType.TOOL, name="konflux:jsonencoded"
+                    ),
+                    annotation_comment=json.dumps(annotation, separators=(",", ":")),
+                )
+            )
+        packages.append(package)
+    return packages, result_annotations
+
+
+async def _extend_spdx_with_base_images(
+    sbom: Document, base_image_refs: list[str | None], base_images: dict[str, Image]
+) -> None:
+    """
+    Extend the SPDX SBOM with the base images.
+    Args:
+        sbom (spdx_tools.spdx.model.Document):
+            SBOM to be edited.
+        base_image_refs (list[str | None]):
+            list of image references, the last one is the parent image.
+        base_images (dict[str, Image]):
+            mapping of those references to Image objects.
+
+    Returns:
+        None: Nothing is returned, changes are performed in-place.
+    """
+    packages, annotations = await _get_spdx_packages_from_base_images(
+        base_image_refs, base_images
+    )
+    if not packages:
+        return
+    sbom.packages.extend(packages)
+    sbom.annotations.extend(annotations)
+    root_spdxids = await find_spdx_root_packages_spdxid(sbom)
+    for root_spdxid in root_spdxids:
+        for package in packages[:-1]:
+            # Those are builder images
+            sbom.relationships.append(
+                Relationship(
+                    spdx_element_id=package.spdx_id,
+                    relationship_type=RelationshipType.BUILD_TOOL_OF,
+                    related_spdx_element_id=root_spdxid,
+                )
+            )
+        # Handle Parent image
+        sbom.relationships.append(
+            Relationship(
```

Comment:
I can't find this relationship in the old script. Where does this requirement come from?

Reply:
This is one of the ACs of this task: implement the requirement from ISV-5708.